discovery-provider/alembic/versions/ec3b20d7bce3_backfill_track_cid.py

```diff
@@ -0,0 +1,145 @@
+"""backfill_track_cid
+
+Revision ID: ec3b20d7bce3
+Revises: 3cdcb5e303f8
+Create Date: 2022-12-08 14:34:59.163989
+
+"""
+import os
+import shutil
+import urllib.request
+import zipfile
+from pathlib import Path
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects.postgresql import ARRAY
+
+# revision identifiers, used by Alembic.
+revision = "ec3b20d7bce3"
+down_revision = "3cdcb5e303f8"
+branch_labels = None
+depends_on = None
+
+
+def copy_mapping_into_temp_table():
+    ####################################################################
+    # Create temporary table and copy track id <> cid mapping into table
+    # Also create index on track_id
+    ####################################################################
+    inner_sql = f"""
+        create table if not exists tmp_track_cid_mapping (track_id varchar unique, track_cid varchar(46));
+        """
+    sql = sa.text("begin; \n\n " + inner_sql + " \n\n commit;")
+    op.get_bind().execute(sql)
+
+    path_tmp = Path(__file__).parent.joinpath("../tmp")
+    path_csv = Path(__file__).parent.joinpath("../tmp/track_cids.csv")
+    path_zip = Path(__file__).parent.joinpath("../tmp/track_cids.csv.zip")
+
+    env = os.getenv("audius_discprov_env")
+    if env != "stage" and env != "prod":
+        return
+    if env == "stage":
+        aws_url = "https://s3.us-west-1.amazonaws.com/download.staging.audius.co/track_cids.csv.zip"
+    else:
+        aws_url = (
+            "https://s3.us-west-1.amazonaws.com/download.audius.co/track_cids.csv.zip"
+        )
+
+    os.mkdir(path_tmp)
+    print(f"Migration - downloading {aws_url}")
+    urllib.request.urlretrieve(aws_url, path_zip)
+    print("Migration - download complete")
+
+    with zipfile.ZipFile(path_zip, "r") as zip_ref:
+        zip_ref.extractall(path_tmp)
+
+    cursor = op.get_bind().connection.cursor()
+    with open(path_csv, "r") as f:
+        # Skip the header row of the csv
+        next(f)
+        cursor.copy_from(
+            f, "tmp_track_cid_mapping", sep=",", columns=("track_id", "track_cid")
+        )
+
+    #############################################################
+    # Change the track id column type to be integer and add index
```

Comment:
why not define column type as integer when creating table?

Reply:
@sddioulde was there a reason you made it a varchar? If not I'll change it to integer

Reply:
if my memory isnt failing me, it failed to copy the track ids from csv as integer. maybe worth trying it again?

Reply:
Nah I figured that would be the case anyway. I'll keep it as is